### PR TITLE
[skip ci] Add connectorTestSuitesOptions to metadata

### PR DESCRIPTION
**Pull request by alafanechere**
## What
Closes https://github.com/airbytehq/airbyte-internal-issues/issues/7549
Closes https://github.com/airbytehq/airbyte-internal-issues/issues/7550

This PR seeds the `connectortTestSuitesOptions` new metadata field on all our connectors

## How
I generated the values with an ad-hoc `airbyte-ci` command (that might not be merged).
I checked test suite existence in the following manner:
#### For python connectors
A connector has unit tests or integration tests if these test folders have files with the `test` substring in it. This is how pytest discovers tests/

#### For java connectors
If a connector has `src/test` I consider they have unit test. If a connector has `src/test-integration` I consider that have integration test

I considered a connector has `acceptanceTest` if they have a file named `acceptance-test-config.yml` in their folder.

#### Secret validation
I pulled secrets from GSM and mapped a secret to a connector thanks to the `connector` secret label.
I ran a `check` with the secret config against a connector to validate that the secret configuration are valid.
Invalid secrets are not declared in the metadata files.

## User Impact
None as it field it not yet used.
They'll be used when work on the following issues will be done:
https://github.com/airbytehq/airbyte-internal-issues/issues/7551
https://github.com/airbytehq/airbyte-internal-issues/issues/7552

## Safe to merge without CI?
Yes I think so. I don't want to run CI on all our connectors as it will not pass on a lot of connectors.
This field introduction is not changing the connector behavior and I manually tested that the metadata files are valid.

@bnchrch merging this will trigger metadata upload of 335 connectors. Just warning you that we might observe an heavy load on Dagster.

### CODE REVIEW

File: airbyte-integrations/connectors/destination-bigquery/metadata.yaml
Comment (bnchrch) [inline airbyte-integrations/connectors/destination-bigquery/metadata.yaml:24]: ❗ were risking that the platform doesnt know how to parse the newlines.

I think thats fine though

File: airbyte-integrations/connectors/destination-snowflake/metadata.yaml
Comment (bnchrch) [inline airbyte-integrations/connectors/destination-snowflake/metadata.yaml:39]: ❓ Wow. This is more individual secrets than I expected. Are we sure they are all actually used? Did we cross reference them with the integration test file?

**Review by bnchrch — APPROVED** [review]
Couple of comments.

The major one being how many secrets we have. Which feels incorrect?

Let me know your thoughts there before you merge.

Otherwise I'm going to approve now so as not to be a blocker

File: airbyte-integrations/connectors/destination-snowflake/metadata.yaml
Comment (alafanechere) [reply airbyte-integrations/connectors/destination-snowflake/metadata.yaml:39]: @bnchrch  Sort of... For instance: https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connectors/destination-snowflake/src/test-integration/java/io/airbyte/integrations/destination/snowflake/typing_deduping/SnowflakeInternalStagingCaseInsensitiveTypingDedupingTest.java#18

Some other are referenced in docs.
If they exists there it means that in any case:
* They pass the `check` call
* They are already fetched and mounted for tests as ci_credentials would have fetched them according to the secret label

File: airbyte-integrations/connectors/destination-bigquery/metadata.yaml
Comment (postamar) [reply airbyte-integrations/connectors/destination-bigquery/metadata.yaml:24]: alternatively, define this `message` field value as a text block, c.f. https://yaml-multiline.info/

File: airbyte-integrations/connectors/destination-snowflake/metadata.yaml
Comment (postamar) [reply airbyte-integrations/connectors/destination-snowflake/metadata.yaml:39]: Correct me if I'm wrong but this seems like something for the destinations team to worry about. FWIW the number of secrets seems plausible to me.

File: airbyte-integrations/connectors/destination-bigquery/metadata.yaml
Comment (alafanechere) [reply airbyte-integrations/connectors/destination-bigquery/metadata.yaml:24]: It was hard to modify yaml and keep ordering, format and comments.
I used `ruamel` and could not find a way to not make it wrap stuff and add new lines...

**Review by postamar — COMMENTED** [review]
I recommend making all `message` fields multiline text blocks, instead of escaping newlines.

File: airbyte-integrations/connectors/destination-snowflake/metadata.yaml
Comment (stephane-airbyte) [reply airbyte-integrations/connectors/destination-snowflake/metadata.yaml:39]: number of secrets is a bit crazy, I'll admit. Not sure why that is.
Do we have doc around the meaning of the alias on a testSecret field? I'm not clear if we modify the way we should add new secrets to a connector either.

Comment (alafanechere) [reply airbyte-integrations/connectors/destination-snowflake/metadata.yaml:39]: @stephane-airbyte 

> number of secrets is a bit crazy, I'll admit. Not sure why that is.

These are all the secrets declared for this connector on GSM at the moment... You can remove unused one from the list safely if they're not used in your tests.

> Do we have doc around the meaning of the alias on a testSecret field? 

Not yet, it's coming. The alias is just a way to not expose the GCP project id where the secret store is. A mapping between aliases and project id will be kept in a. GHA secret.

The slight change in the way we add secret to a connector is that following the creation of a secret in GSM you'd have to update the metadata file to declare a pointer to this secret so that `airbyte-ci` will fetch them .

Comment (stephane-airbyte) [reply airbyte-integrations/connectors/destination-snowflake/metadata.yaml:39]: I'm mostly asking because I have a PR in flight that adds a new secret to destination-s3: https://github.com/airbytehq/airbyte/pull/38204
The secrets are already in GSM, so you can add them in this PR as well

File: airbyte-integrations/connectors/source-salesforce/metadata.yaml
Comment (bazarnov) [inline airbyte-integrations/connectors/source-salesforce/metadata.yaml:56]: This is not used anywhere for the `source-salesforce`

File: airbyte-integrations/connectors/source-salesforce/metadata.yaml
Comment (bazarnov) [inline airbyte-integrations/connectors/source-salesforce/metadata.yaml:61]: In `CAT`, the `config.json` and `config_sandbox.json` are used and fetched from GSM.

File: airbyte-integrations/connectors/source-salesforce/metadata.yaml
Comment (bazarnov) [inline airbyte-integrations/connectors/source-salesforce/metadata.yaml:39]: Not used.

File: airbyte-integrations/connectors/source-shopify/metadata.yaml
Comment (bazarnov) [inline airbyte-integrations/connectors/source-shopify/metadata.yaml:108]: All good here.

File: airbyte-integrations/connectors/source-hubspot/metadata.yaml
Comment (bazarnov) [inline airbyte-integrations/connectors/source-hubspot/metadata.yaml:84]: There are 2 cases for `config_oauth_no_start_date.json`, but we should probably include only 1 of them. I'm not sure which exactly. Maybe @maxi297 or @strosek could advise here?

**Review by bazarnov — APPROVED** [review]
Looks good to me, i'd like to highlight the fact, that some of the CREDS in GSM are either outdated or not used anymore, it's hard to say confidently which ones are fine to remove, it requires manual check for each source and know what secrets exactly is pulled/pulled and override the previously pulled one, etc

Let's merge, so this is not blocked on us, but when any source or destination is under future maintenance - the engineer should be able to validate the secrets pulled and clean the broken ones / duplicated once from the `metadata.yaml`

**Comment by alafanechere** [comment]
> Looks good to me, i'd like to highlight the fact, that some of the CREDS in GSM are either outdated or not used anymore, it's hard to say confidently which ones are fine to remove, it requires manual check for each source and know what secrets exactly is pulled/pulled and override the previously pulled one, etc
> 
> Let's merge, so this is not blocked on us, but when any source or destination is under future maintenance - the engineer should be able to validate the secrets pulled and clean the broken ones / duplicated once from the `metadata.yaml`

💯 agree @bazarnov  The single thing I know for sure is that any config listed here are passing the `check` command, but it does not mean it's actually use.

**Comment by octavia-approvington** [comment]
Myomoto says it looks good <br/> thats a niiice

**Review by octavia-approvington — APPROVED** [review]

**Pull request merged**
